src/computational_graph/tree_properties.jl

```diff
@@ -0,0 +1,58 @@
+##################### AbstractTrees interface for Graphs ########################### 
+
+## Things that make printing prettier
+AbstractTrees.printnode(io::IO, g::Graph) = print(io, "\u001b[32m$(g.id)\u001b[0m : $g")
+
+## Guarantee type-stable tree iteration for Graphs
+AbstractTrees.NodeType(::Graph) = HasNodeType()
+AbstractTrees.nodetype(::Graph{F,W}) where {F,W} = Graph{F,W}
+
+## Optional enhancements
+# These next two definitions allow inference of the item type in iteration.
+# (They are not sufficient to solve all internal inference issues, however.)
+Base.IteratorEltype(::Type{<:TreeIterator{Graph{F,W}}}) where {F,W} = Base.HasEltype()
+Base.eltype(::Type{<:TreeIterator{Graph{F,W}}}) where {F,W} = Graph{F,W}
+
+function AbstractTrees.children(g::Graph)
+    return g.subgraphs
+end
+
+##################### Tree properties ########################### 
+
+# Does the graph have any children?
```

Comment:
are they external functions? If so, you need to add doc strings.
"""
      function ....

"""

Reply:
I think we only need these helper functions internally, but I can add docstrings nonetheless.

Reply:
Done. I also exported these functions for now; we can decide later if we want to move them to internal-only.